common/src/types/v0/transport/replica.rs

```diff
@@ -21,6 +22,103 @@ impl GetReplicas {
+    /// Return a new `Self` from the provided arguments
+    pub fn new(name: &PoolId, uuid: Option<PoolUuid>) -> Self {
+        Self {
+            name: name.clone(),
+            uuid,
+        }
+    }
+    /// get the uuid of the pool
+    pub fn get_uuid(&self) -> Result<PoolUuid, uuid::Error> {
+        match self.uuid.clone() {
+            Some(uuid) => Ok(uuid),
+            None => {
+                // TODO add a getter function to fetch the uuid of the pool
+                Ok(PoolUuid::try_from(self.name.to_string())?)
+            }
+        }
+    }
+
+    /// get name of the pool from uuid
+    pub fn get_name_from_uuid(_uuid: &PoolUuid) -> PoolId {
+        unimplemented!()
+    }
+}
+
+impl<'de> Deserialize<'de> for ReplicaPool {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let json: Value = Value::deserialize(deserializer)?;
+        println!("json: {:?}", json);
+
+        let replica_pool = match json {
+            Value::Object(pool) => {
+                let name = match pool.get(&"name".to_string()) {
+                    Some(name) => match PoolId::try_from(name) {
+                        Ok(name) => name,
+                        Err(e) => return Err(serde::de::Error::custom(e)),
+                    },
+                    None => {
+                        return Err(serde::de::Error::custom(
+                            "Missing name field from the replica pool".to_string(),
+                        ))
+                    }
+                };
+                let uuid = match pool.get(&"uuid".to_string()) {
+                    Some(uuid) => match uuid.as_null() {
+                        Some(()) => None,
+                        None => match PoolUuid::try_from(uuid) {
+                            Ok(uuid) => Some(uuid),
+                            Err(e) => return Err(serde::de::Error::custom(e)),
+                        },
+                    },
+                    None => {
+                        return Err(serde::de::Error::custom(
+                            "Missing uuid field from the replica pool".to_string(),
+                        ))
+                    }
+                };
+                ReplicaPool::new(&name, uuid)
+            }
+            Value::String(name) => ReplicaPool::new(&name.into(), None),
+            _ => return Err(serde::de::Error::custom("Invalid replica pool".to_string())),
+        };
+
+        Ok(replica_pool)
+    }
+}
+
+#[cfg(test)]
+mod tests_de {
+    use super::*;
+
+    #[test]
+    fn test_replica_pool_deserializer() {
+        let inputs: Vec<&str> = vec![
+            r#"{"name":"pool-1", "uuid":null}"#,
+            r#"{"name":"pool-1", "uuid":"22ca10d3-4f2b-4b95-9814-9181c025cc1a"}"#,
+        ];
+
+        for input in &inputs {
+            let replica_pool: ReplicaPool = serde_json::from_str(input).unwrap();
+            println!("{:?}", replica_pool);
```

Comment:
would be good to use asserts to make sure the values are expected.
We're also missing a test case for the string case.

Reply:
Custom deserializer for ReplicaPool has been removed hence this test also.